### PR TITLE
5164 - validation

### PR DESCRIPTION
**Pull request by codyseibert**
Working on the various validation logic for this story.

### CODE REVIEW

File: shared/src/business/useCases/validateAddDeficiencyStatisticsInteractor.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const { Statistic } = require('../entities/Statistic');
+
+/**
+ * validatePetitionFromPaper
+ *
+ * @param {object} providers the providers object
+ * @param {object} providers.applicationContext the application context
+ * @param {object} providers.petition the petition data to validate
+ * @returns {object} errors (null if no errors)
```
Comment (rachaelparris) [inline shared/src/business/useCases/validateAddDeficiencyStatisticsInteractor.js]: jsdocs

File: shared/src/business/entities/Statistic.js [hunk 2]
```diff
@@ -28,30 +28,57 @@ function Statistic(rawStatistic) {
...
 joiValidationDecorator(
   Statistic,
   joi.object().keys({
     determinationDeficiencyAmount: joi
-      .number()
-      .optional()
-      .allow(null)
-      .description('The amount of the deficiency determined by the Court.'),
+      .alternatives()
+      .conditional('determinationTotalPenalties', {
```
Comment (rachaelparris) [inline shared/src/business/entities/Statistic.js:50]: Could you add some tests for these?

File: shared/src/business/useCases/validateAddDeficiencyStatisticsInteractor.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+const { Statistic } = require('../entities/Statistic');
+
+/**
+ * validatePetitionFromPaper
+ *
+ * @param {object} providers the providers object
+ * @param {object} providers.applicationContext the application context
+ * @param {object} providers.petition the petition data to validate
```
Comment (matthopson) [inline shared/src/business/useCases/validateAddDeficiencyStatisticsInteractor.js]: statistic

File: web-client/src/applicationContext.js [hunk 4]
```diff
@@ -86,6 +86,7 @@ import {
   getFormattedCaseDetail,
   sortDocketRecords,
 } from '../../shared/src/business/utilities/getFormattedCaseDetail';
+import { validateAddDeficiencyStatisticsInteractor } from '../../shared/src/business/useCases/validateAddDeficiencyStatisticsInteractor';
```
Comment (rachaelparris) [inline web-client/src/applicationContext.js]: Nit: could you combine these blocks that are separated by a newline, so they're sorted correctly?

File: web-client/src/presenter/actions/setDefaultFormForAddDeficiencySatisticsAction.js [hunk 5]
```diff
@@ -6,6 +6,11 @@ import { state } from 'cerebral';
...
  * @param {object} providers.store the cerebral store used for setting state.workItem
  */
-export const setDefaultFormForAddDeficiencySatisticsAction = ({ store }) => {
-  store.set(state.form.yearOrPeriod, 'Year');
+export const setDefaultFormForAddDeficiencySatisticsAction = ({
+  props,
+  store,
+}) => {
+  store.set(state.form, {
+    yearOrPeriod: props.yearOrPeriod || 'Year',
```
Comment (rachaelparris) [inline web-client/src/presenter/actions/setDefaultFormForAddDeficiencySatisticsAction.js]: When is there a need to change the default here?

File: web-client/src/presenter/actions/validateAddDeficiencyStatisticsAction.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+import { state } from 'cerebral';
+
+/**
+ * validates the petition.
+ *
+ * @param {object} providers the providers object
+ * @param {object} providers.applicationContext the application context needed for getting the validatePetition use case
+ * @param {object} providers.get the cerebral get function used for getting state.form
+ * @param {object} providers.path the cerebral path which contains the next path in the sequence (path of success or error)
+ * @returns {object} the next path based on if validation was successful or error
```
Comment (rachaelparris) [inline web-client/src/presenter/actions/validateAddDeficiencyStatisticsAction.js]: jsdocs

File: shared/src/business/useCases/validateAddDeficiencyStatisticsInteractor.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+const { Statistic } = require('../entities/Statistic');
+
+/**
+ * validatePetitionFromPaper
```
Comment (matthopson) [inline shared/src/business/useCases/validateAddDeficiencyStatisticsInteractor.js]: jsdoc

File: web-client/src/views/CaseDetail/AddDeficiencyStatistics.jsx [hunk 8]
```diff
@@ -58,10 +66,14 @@ export const AddDeficiencyStatistics = connect(
                         type="radio"
                         value={option}
                         onChange={e => {
+                          setupAddDeficiencyStatisticsFormSequence({
```
Comment (rachaelparris) [inline web-client/src/views/CaseDetail/AddDeficiencyStatistics.jsx]: I think here you may want something similar to `updateStatisticsFormValueSequence`, which sets the value and then clears the form

File: web-client/src/presenter/actions/clearAddDeficiencyFormValuesAction.js [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+ * @param {object} providers.store the cerebral store
+ * @param {object} providers.props the cerebral props object
+ */
+export const clearAddDeficiencyFormValuesAction = ({ props, store }) => {
+  if (props.key.includes('yearOrPeriod')) {
+    store.set(state.form, {
+      yearOrPeriod: props.value,
+    });
+  }
+};
```
Comment (matthopson) [inline web-client/src/presenter/actions/clearAddDeficiencyFormValuesAction.js:16]: Test

File: web-client/src/presenter/actions/setDefaultFormForAddDeficiencySatisticsAction.js (hunk 5 above)
Comment (codyseibert) [reply web-client/src/presenter/actions/setDefaultFormForAddDeficiencySatisticsAction.js]: it's not needed anymore, I'll set it back to Year

**Review by matthopson — APPROVED** [review]

**Review by rachaelparris — APPROVED** [review]

**Pull request merged**